src/test/java/com/aws/greengrass/clientdevices/auth/iot/registry/CertificateRegistryTest.java

```diff
@@ -5,114 +5,116 @@
+import com.aws.greengrass.clientdevices.auth.certificate.CertificateHelper;
+import com.aws.greengrass.clientdevices.auth.certificate.CertificateStore;
+import com.aws.greengrass.clientdevices.auth.helpers.CertificateTestHelpers;
+import com.aws.greengrass.clientdevices.auth.iot.Certificate;
+import com.aws.greengrass.clientdevices.auth.iot.InvalidCertificateException;
 import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Captor;
-import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
 import java.util.Optional;
 
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
-import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.reset;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
 
 @ExtendWith({MockitoExtension.class, GGExtension.class})
 class CertificateRegistryTest {
-    private static final String mockCertPem = "certificatePem";
-    private static final String mockCertId = "certificateId";
-    @Mock
-    private IotAuthClient mockIotAuthClient;
-    @Captor
-    private ArgumentCaptor<String> certPemCaptor;
+    private static X509Certificate validClientCertificate;
+    private static String validClientCertificatePem;
 
     private CertificateRegistry registry;
 
+    @BeforeAll
+    static void beforeAll() {
+        try {
+            KeyPair rootKeyPair = CertificateStore.newRSAKeyPair(2048);
+            KeyPair clientKeyPair = CertificateStore.newRSAKeyPair(2048);
+            X509Certificate rootCA = CertificateTestHelpers.createRootCertificateAuthority("root", rootKeyPair);
+
+            validClientCertificate = CertificateTestHelpers.createClientCertificate(rootCA, "Client", clientKeyPair.getPublic(),
+                    rootKeyPair.getPrivate());
+            validClientCertificatePem = CertificateHelper.toPem(validClientCertificate);
+        } catch (NoSuchAlgorithmException | CertificateException | OperatorCreationException | IOException e) {
+            Assertions.fail("Unable to initialize test certificates!");
+        }
```

Comment:
just let this method throw, it will do the same thing, without swallowing the exception